### DISCUSSION TITLE
Need help & suggestions for integrating sunbird content player

### DISCUSSION THREAD

**Discussion by snehal0904** [Sunbird-Knowlg/Community#81] [Q&A] [answered]
Hello Team,

We are working on one project, where we want to show a courses list as per student performance and hence we have built a custom frontend to display a list of courses and play those. 

We have successfully integrated, QUML, PDF, and video players and can open and track content in these players.
Now, we want to play h5p content, so we are trying to integrate a content player with our application. We have integrated this player but as we are using sunbird saas / Diksha as a content library (Course content is getting created here and using content get APIs we are fetching details) we are getting CORS policy errors when we are trying to play the h5p content.

I am attaching the error screenshot we are getting while playing this, please suggest possible solutions for this.
@Krishnaj20 @pallakartheekreddy @coolbung @parthlawate @itsvick @vinukumar-vs 

![Screenshot from 2022-11-28 09-54-35](https://user-images.githubusercontent.com/17213411/204449047-4e77e9be-9f6b-4348-8f8d-4627f9056081.png)

**Comment by vinukumar-vs** [comment, marked as answer]
@snehal0904 This is not specific to player. The browser is not allowing to access the data from different host, hence it is throwing the CORS issue. 

Primary host(portal launch URL): "https://alt-shiksha.uniteframework.io"
Secondary host(content host): "https://druva.shikshalokam.org"

There are few solutions to handle this
1. Have a proxy layer which will call with the same host but poxy will redirect to actual host.
2. All permission at seconday(content) host level to access the primary host address(Enabling CORS at blob level)
3. upload the content to the same host & access with the same host URL.

 @coolbung @parthlawate

**Reply by snehal0904** [reply]
@vinukumar-vs Thank you for the response. 

Yes, this is not player specific. This is because h5p content required some JS files to be called and as we are playing the content on the different hosts we are facing this issue.

Have a proxy layer which will call with the same host but poxy will redirect to actual host.
 - As the player library is making these calls, making changes in the player code might be not that straightforward. We will check this solution.
 
All permission at seconday(content) host level to access the primary host address(Enabling CORS at blob level)
- Yes, we are checking this option. But we can implement this option in the case of the SAAS instance only. If we want want to use Diksha as a content host/library, will it be possible for Diksha to allow this permission to play the content on other hosts?

upload the content to the same host & access with the same host URL.
- As the application demands, performance-based unlocking of the content accessing the content from the SAAS instance is not possible.

**Reply by snehal0904** [reply]
We have taken the first proxy option, able to play the h5p content on our application. Thank you @vinukumar-vs
